homotopy-core/src/diagram.rs

```diff
@@ -177,7 +177,11 @@ thread_local! {
 }
 
 impl DiagramN {
-    pub fn new<S, T>(generator: Generator, source: S, target: T) -> Result<Self, NewDiagramError>
+    pub fn from_generator<S, T>(
```

Comment:
Maybe this should be a `impl<S: Into<Diagram>, T: Into<Diagram>> TryFrom<(Generator, S, T)> for DiagramN`?